src/tracking-buffer/writable-tracking-buffer.js

```diff
@@ -1,27 +1,40 @@
+// @flow
+
 const bigint = require('./bigint');
 
 const SHIFT_LEFT_32 = (1 << 16) * (1 << 16);
 const SHIFT_RIGHT_32 = 1 / SHIFT_LEFT_32;
 const UNKNOWN_PLP_LEN = new Buffer([0xfe, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff]);
 const ZERO_LENGTH_BUFFER = new Buffer(0);
 
+export type Encoding = 'utf8' | 'ucs2' | 'ascii';
+
 /*
   A Buffer-like class that tracks position.
 
   As values are written, the position advances by the size of the written data.
   When writing, automatically allocates new buffers if there's not enough space.
  */
 module.exports = class WritableTrackingBuffer {
-  constructor(initialSize, encoding, doubleSizeGrowth) {
+  initialSize: number;
+  encoding: Encoding;
+  doubleSizeGrowth: boolean;
+
+  buffer: Buffer;
+  compositeBuffer: Buffer;
+
+  position: number;
+
+  constructor(initialSize: number, encoding: ?Encoding, doubleSizeGrowth: ?boolean) {
     this.initialSize = initialSize;
-    this.encoding = encoding;
-    this.doubleSizeGrowth = doubleSizeGrowth;
-    this.doubleSizeGrowth || (this.doubleSizeGrowth = false);
-    this.encoding || (this.encoding = 'ucs2');
+    this.encoding = encoding || 'ucs2';
+    this.doubleSizeGrowth = doubleSizeGrowth || false;
     this.buffer = new Buffer(this.initialSize);
+    this.compositeBuffer = new Buffer(0);
```

Comment:
I changed the `compositeBuffer` to be initialized in the constructor. This simplifies the `newBuffer` method a tiny bit and keeps us from having to check whether `this.compositeBuffer` is `undefined` or not before use.